### PR TITLE
CI: Replace autopep8 & flake8 by ruff for linting backend code

### PR DESCRIPTION
**Pull request by drikusroor**
This pull request replaces the autopep8 and flake8 linters with ruff for linting the backend code. It also adds a script to format the backend code using ruff. The default `ruff.toml` configuration file is included, and the linting script now allows parameters. Additionally, the documentation recommends using the vscode ruff extension with this project.

### CODE REVIEW

File: backend/ruff.toml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    "node_modules",
+    "site-packages",
+    "venv",
+]
+
+# Set line length to match flake8 configuration
+line-length = 120
+indent-width = 4
+
+# Assume Python 3.8
```
Comment (Evert-R) [inline backend/ruff.toml:35]: We are using python 3.11.9 at the moment

**Review by Evert-R — CHANGES_REQUESTED** [review]
Nice. Only change the python version to 3.11

File: backend/ruff.toml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+target-version = "py38"
```
Comment (drikusroor) [inline backend/ruff.toml:36]:
```suggestion
target-version = "py311"
```

File: backend/ruff.toml (hunk 1 above)
Comment (drikusroor) [reply backend/ruff.toml:35]:
```suggestion
# Assume Python 3.11
```

File: scripts/lint-back [hunk 3]
```diff
@@ -1,2 +1,8 @@
 #!/bin/bash
-docker-compose run --rm server bash -c "flake8"
+
+# Ruff allows you to lint your Python code with a variety of linters.
+# This script runs the Ruff linter on the backend codebase.
+# It uses the `ruff check` command to lint the codebase.
+# The ` --watch` flag tells Ruff to watch the codebase for changes and re-lint the codebase when changes are detected.
```
Comment (drikusroor) [inline scripts/lint-back:6]:
```suggestion
# The `--watch` flag tells Ruff to watch the codebase for changes and re-lint the codebase when changes are detected.
```

**Pull request merged**